### ISSUE TITLE
Content of tabs not preserved when showing/hiding SubLayout

### ISSUE THREAD

**Issue by marekgregor** [caplin/FlexLayout#121]
[This fix](https://github.com/caplin/FlexLayout/commit/18041caa67ea290918d3fe76a838205fef3949e2#diff-25d902c24283ab8cfbac54dfa101ad31) of Layout.tsx file (introduced in the version 0.3.10) unfortunately caused another issue in our case.

When we have Layout in Layout (SubLayout), hiding the content of tab with SubLayout cause that current content of SubLayout is completely lost because of the condition `if (this.rect.width === 0)` in the fix, which does not pass only when the SubLayout is rendered initially, but always when SubLayout is hidden.

The solution is to introduce a new flag and change the condition  to `if (this.widthWasNonZero)`, this new flag will be set up in updateRect method to true when the width is bigger than 0: `
if(this.domRect.width > 0) {
this.widthWasNonZero = true;
}
`

thanks

**Comment by nealus** [comment]
I'm not sure that zero check is needed at all:
```
    // first render will be used to find the size
    if (this.rect.width === 0) { 
      return (
        <div ref={this.selfRef}
          className={this.getClassName("flexlayout__layout")} />
      );
    }
```
can you try simply removing it (removed from latest version and cannot see any problems)

**Comment by nealus** [comment]
Just published 0.4.4 which should fix the problem

**Comment by marekgregor** [comment]
Amazing, it is working like a charm. Thanks a lot.
